### PR TITLE
Added DefinitionResolver

### PR DESCRIPTION
**Pull request by mcop1**
## Changes in this pull request
Part of https://pimcore.atlassian.net/browse/PSDP-174

### WHAT
Added DefinitionResolver to add or retrieve permissions

### CODE REVIEW

File: src/Models/User/Permission/DefinitionResolver.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\StaticResolverBundle\Models\User\Permission;
+
+use Pimcore\Model\User\Permission;
+
```
Comment (mattamon) [inline src/Models/User/Permission/DefinitionResolver.php:21]: Concrete class can be internal and final

File: src/Models/User/Permission/DefinitionResolverInterface.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+declare(strict_types=1);
+
```
Comment (mattamon) [inline src/Models/User/Permission/DefinitionResolverInterface.php:3]: License is missing

File: src/Models/User/Permission/DefinitionResolver.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+class DefinitionResolver implements DefinitionResolverInterface
+{
+    /**
+     * @throws Exception
+     */
+    public function getByKey(string $permission): ?Permission\Definition
+    {
+        return Permission\Definition::getByKey($permission);
```
Comment (mattamon) [inline src/Models/User/Permission/DefinitionResolver.php]: Can we just import the Definition here instead of Permission\Definition?

**Review by mattamon — CHANGES_REQUESTED** [review]

**Comment by mcop1** [comment]
Too fast too furious pr reviewing starring @mattamon
But thanks for your comments of course :) :+1:

**Comment by mattamon** [comment]
> Too fast too furious pr reviewing starring @mattamon But thanks for your comments of course :) 👍

Without Family, You've Got Nothing.
You'll Always be With Me, and You'll Always be My Brother.
I Don’t Have Friends, I’ve Got Family

**Review by mattamon — APPROVED** [review]
No matter where or what you code you will always be family.

**Comment by lukmzig** [comment]
> No matter where or what you code you will always be family.

… It's been a long day without you, my friend
And I'll tell you all about it when I see you again
We've come a long way from where we began
Oh, I'll tell you all about it when I see you again
When I see you again

**Pull request merged**